src/vals/machine_eq.rs

```diff
@@ -0,0 +1,6 @@
+pub trait MachineEq {
+    type Pred;
+
+    fn machine_eq(&self, other: &Self) -> Self::Pred;
+    fn machine_ite(pred: Self::Pred, then: Self, xelse: Self) -> Self;
+}
```

Comment:
Abstract interface for any type in the machine that can be transformed based off of an equality